simplify net-related code a bit and improve some other things

codesenberg opened the pull request:
In [this](https://www.reddit.com/r/golang/comments/61y525/would_love_some_feedbackcritique_of_my_first_cli/) thread you asked for feedback/critique, so, I'm sending you this PR to implement everything suggested by me and [/u/allhatenocattle](https://www.reddit.com/user/allhatenocattle), plus a bit more.
Not entirely sure about correctness of net-related changes, but they look OK to me.

codesenberg committed:
simplify net-related code a bit and improve some other things

robphoenix commented on trawl.go:
```diff
@@ -31,81 +31,54 @@ func New(iface net.Interface) (i *Interface, err error) {
+			if ip := ipnet.IP.To4(); ip != nil {
+				ipv4 = ipnet
+			} else {
+				ipv6 = ipnet
```
This blows up if there's no IPv6 address, when the `String()` method is called on line 46 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x485696]

goroutine 1 [running]:
panic(0x603ea0, 0xc42000c1f0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
net.networkNumberAndMask(0x0, 0x3, 0xc42000d360, 0xd, 0x4, 0x4, 0x3)
	/usr/local/go/src/net/ip.go:443 +0x26
net.(*IPNet).String(0x0, 0xc42000d360, 0xd)
	/usr/local/go/src/net/ip.go:493 +0x40
main.New(0x3, 0x5dc, 0xc42000d1a0, 0x7, 0xc4200d49f0, 0x6, 0xf10, 0x13, 0xc4200180e0, 0x0, ...)
	/home/rob/code/go/src/github.com/robphoenix/trawl/trawl.go:47 +0x17c
main.main()
	/home/rob/code/go/src/github.com/robphoenix/trawl/main.go:62 +0x2ca
```

codesenberg replied:
Yep, my bad. Will fix this soon.

robphoenix commented on trawl.go:
```diff
@@ -31,81 +31,54 @@ func New(iface net.Interface) (i *Interface, err error) {
+}
 
-	return strings.Join(mask, "."), nil
+func maskedIPString(ipnet *net.IPNet) string {
```
I'm not sure I understand what advantages this brings over the current implementation?

```
// get IPv4 network
_, ipnet, err := net.ParseCIDR(ipv4)
if err != nil {
	return i, err
}
```

codesenberg replied:
Well, it's more straightforward and simple, if you ask me. Is it not?

robphoenix replied:
I don't know, in my mind it seems to be extra work than is necessary, when `ParseCIDR()` does it for you.

robphoenix commented:
This is amazing! Thankyou @codesenberg for taking the time to do this, so much cleaner and I've learnt a few things also. :tada: 
I have a couple of comments. I'm happy to merge and fix up the `panic` when there's no IPv6 address issue myself if you don't have the time to update the PR, let me know.
:+1:

codesenberg committed:
trawl.go: fix bug found out by @robphoenix during review

Now code correctly handles situations, when interface doesn't have IPv6
address, instead of panicking.

codesenberg commented:
@robphoenix, all done.

robphoenix commented:
Super, thanks @codesenberg really appreciate this.
